### ISSUE TITLE
Implement DQ Step Through STCAL

### ISSUE THREAD

**Issue by stscijgbot-rstdms** [spacetelescope/romancal#189]
_Issue [RCAL-137](https://jira.stsci.edu/browse/RCAL-137) was created on JIRA by [Paul Huwe](https://jira.stsci.edu/secure/ViewProfile.jspa?name=phuwe):_

Implement DQ step via STCAL.

**Comment by stscijgbot-rstdms** [comment]
_Comment by [Paul Huwe](https://jira.stsci.edu/secure/ViewProfile.jspa?name=phuwe) on [JIRA](https://jira.stsci.edu/browse/RCAL-137?focusedCommentId=518058#comment-518058):_

I have an initial crack at this, but stcal has been recently updated and it is clashing with my work. I will coordinate with [David Davis](https://jira.stsci.edu/secure/ViewProfile.jspa?name=ddavis) and [Nadia Dencheva](https://jira.stsci.edu/secure/ViewProfile.jspa?name=dencheva) about it tomorrow.

**Comment by stscijgbot-rstdms** [comment]
_Comment by [Paul Huwe](https://jira.stsci.edu/secure/ViewProfile.jspa?name=phuwe) on [JIRA](https://jira.stsci.edu/browse/RCAL-137?focusedCommentId=518305#comment-518305):_

I have updated the DQ Datamodel code to utilize STCAL. This is available this branch in my fork: <https://github.com/PaulHuwe/romancal/tree/RCAL-137_DQviaSTCAL>

I have started porting over the DQ step code that I had worked on before to STCAL. I am in the middle of making that code work in STCAL and link to RomanCAL.

**Comment by stscijgbot-rstdms** [comment]
_Comment by [Paul Huwe](https://jira.stsci.edu/secure/ViewProfile.jspa?name=phuwe) on [JIRA](https://jira.stsci.edu/browse/RCAL-137?focusedCommentId=518493#comment-518493):_

An initial implementation can be found in the following branches of the RCAL & STCAL repos:

RomanCAL: <https://github.com/PaulHuwe/romancal/tree/RCAL-137_DQviaSTCAL>
STCAL: <https://github.com/PaulHuwe/stcal/tree/RCAL-137_DQviaSTCAL>

**Comment by stscijgbot-rstdms** [comment]
_Comment by [Paul Huwe](https://jira.stsci.edu/secure/ViewProfile.jspa?name=phuwe) on [JIRA](https://jira.stsci.edu/browse/RCAL-137?focusedCommentId=518564#comment-518564):_

I misunderstood closing status - reopening due to lack of PR.

**Comment by stscijgbot-rstdms** [comment]
_Comment by [Paul Huwe](https://jira.stsci.edu/secure/ViewProfile.jspa?name=phuwe) on [JIRA](https://jira.stsci.edu/browse/RCAL-137?focusedCommentId=520381#comment-520381):_

This ticket is paused until my RAD work (and possible roman_datamodels work) is completed.

**Comment by stscijgbot-rstdms** [comment]
_Comment by [Paul Huwe](https://jira.stsci.edu/secure/ViewProfile.jspa?name=phuwe) on [JIRA](https://jira.stsci.edu/browse/RCAL-137?focusedCommentId=528078#comment-528078):_

I have gotten the code as far as I can without support for several datamodels in roman_datamodels. This ticket is blocked until this support is implemented.

NOTE: the RAD work needed is completed in RAD PR-23 spacetelescope/rad#23

**Comment by stscijgbot-rstdms** [comment]
_Comment by [Paul Huwe](https://jira.stsci.edu/secure/ViewProfile.jspa?name=phuwe) on [JIRA](https://jira.stsci.edu/browse/RCAL-137?focusedCommentId=535915#comment-535915):_

Accidental closure.

**Comment by stscijgbot-rstdms** [comment]
_Comment by [Paul Huwe](https://jira.stsci.edu/secure/ViewProfile.jspa?name=phuwe) on [JIRA](https://jira.stsci.edu/browse/RCAL-137?focusedCommentId=536392#comment-536392):_

I have continued work on this, which spans several libraries (and has spawned some smaller tickets).

**Comment by stscijgbot-rstdms** [comment]
_Comment by [Paul Huwe](https://jira.stsci.edu/secure/ViewProfile.jspa?name=phuwe) on [JIRA](https://jira.stsci.edu/browse/RCAL-137?focusedCommentId=537233#comment-537233):_

This work is implemented in the following PRs:
#252

spacetelescope/stcal#34

spacetelescope/roman_datamodels#17

spacetelescope/rad#61

 

There are a few specific parts that are blocked, and are being removed to separate tickets, but the bulk of the work is done here.

**Comment by stscijgbot-rstdms** [comment]
_Comment by [David Davis](https://jira.stsci.edu/secure/ViewProfile.jspa?name=ddavis) on [JIRA](https://jira.stsci.edu/browse/RCAL-137?focusedCommentId=538751#comment-538751):_

SInce the PR is still open this should also be open. Once the PR is approved and merged this can be closed.

**Comment by stscijgbot-rstdms** [comment]
_Comment by [Paul Huwe](https://jira.stsci.edu/secure/ViewProfile.jspa?name=phuwe) on [JIRA](https://jira.stsci.edu/browse/RCAL-137?focusedCommentId=539361#comment-539361):_

Status update on the PRs in this ticket:

RAD 61: This is blocked until INS makes a decision about dynamic dq support.

RDM 17: This is blocked until INS makes a decision about dynamic dq support.

STCAL 34: This is closed, and I will be making a new PR, following the proper STCAL procedure.

RCAL 252: This is blocked until the other three are updated.

**Comment by stscijgbot-rstdms** [comment]
_Comment by [Paul Huwe](https://jira.stsci.edu/secure/ViewProfile.jspa?name=phuwe) on [JIRA](https://jira.stsci.edu/browse/RCAL-137?focusedCommentId=539494#comment-539494):_

Status update on the PRs in this ticket:

RAD 61: This has been unblocked, and work is resuming. (Removing dq_def and addressing PR comments.)

RDM 17: This has been unblocked, and work is resuming. (Removing dq_def and dynamic dq code.)

**Comment by stscijgbot-rstdms** [comment]
_Comment by [Paul Huwe](https://jira.stsci.edu/secure/ViewProfile.jspa?name=phuwe) on [JIRA](https://jira.stsci.edu/browse/RCAL-137?focusedCommentId=539932#comment-539932):_

The RAD & RDM tickets are back in PR with all requested changes made. Romancal work continues - most of the tests and code are working, I am finalizing the strun step work.

**Comment by stscijgbot-rstdms** [comment]
_Comment by [Paul Huwe](https://jira.stsci.edu/secure/ViewProfile.jspa?name=phuwe) on [JIRA](https://jira.stsci.edu/browse/RCAL-137?focusedCommentId=540388#comment-540388):_

The RAD & RDM PRs have been merged.

Romancal code has a step producing output which is under review.

**Comment by stscijgbot-rstdms** [comment]
_Comment by [Paul Huwe](https://jira.stsci.edu/secure/ViewProfile.jspa?name=phuwe) on [JIRA](https://jira.stsci.edu/browse/RCAL-137?focusedCommentId=540828#comment-540828):_

The romancal work is nor implemented in RCAL PR 262:
#262

**Comment by stscijgbot-rstdms** [comment]
_Comment by [Paul Huwe](https://jira.stsci.edu/secure/ViewProfile.jspa?name=phuwe) on [JIRA](https://jira.stsci.edu/browse/RCAL-137?focusedCommentId=541721#comment-541721):_

I am addressing a PR comment regarding RampModel instantiation.

**Comment by stscijgbot-rstdms** [comment]
_Comment by [Paul Huwe](https://jira.stsci.edu/secure/ViewProfile.jspa?name=phuwe) on [JIRA](https://jira.stsci.edu/browse/RCAL-137?focusedCommentId=542311#comment-542311):_

All PR comments have been addressed, and ScienceRaw support is now in. This is back in review.

**Comment by stscijgbot-rstdms** [comment]
_Comment by [Paul Huwe](https://jira.stsci.edu/secure/ViewProfile.jspa?name=phuwe) on [JIRA](https://jira.stsci.edu/browse/RCAL-137?focusedCommentId=543608#comment-543608):_

Approved and merged.
